gcp_variant_transforms/libs/bigquery_vcf_schema.py

```diff
@@ -84,130 +41,89 @@ class _TableFieldConstants(object):
-      mode=_TableFieldConstants.MODE_NULLABLE,
+      name=bigquery_util.ColumnKeyConstants.END_POSITION,
+      type=bigquery_util.TableFieldConstants.TYPE_INTEGER,
+      mode=bigquery_util.TableFieldConstants.MODE_NULLABLE,
       description=('End position (0-based). Corresponds to the first base '
                    'after the last base in the reference allele.')))
   schema.fields.append(bigquery.TableFieldSchema(
-      name=ColumnKeyConstants.REFERENCE_BASES,
-      type=_TableFieldConstants.TYPE_STRING,
-      mode=_TableFieldConstants.MODE_NULLABLE,
+      name=bigquery_util.ColumnKeyConstants.REFERENCE_BASES,
+      type=bigquery_util.TableFieldConstants.TYPE_STRING,
+      mode=bigquery_util.TableFieldConstants.MODE_NULLABLE,
       description='Reference bases.'))
 
-  # Add alternate bases.
-  alternate_bases_record = bigquery.TableFieldSchema(
-      name=ColumnKeyConstants.ALTERNATE_BASES,
-      type=_TableFieldConstants.TYPE_RECORD,
-      mode=_TableFieldConstants.MODE_REPEATED,
-      description='One record for each alternate base (if any).')
-  alternate_bases_record.fields.append(bigquery.TableFieldSchema(
-      name=ColumnKeyConstants.ALTERNATE_BASES_ALT,
-      type=_TableFieldConstants.TYPE_STRING,
-      mode=_TableFieldConstants.MODE_NULLABLE,
-      description='Alternate base.'))
-  if split_alternate_allele_info_fields:
-    for key, field in header_fields.infos.iteritems():
-      if field.num == vcf.parser.field_counts[_FIELD_COUNT_ALTERNATE_ALLELE]:
-        alternate_bases_record.fields.append(bigquery.TableFieldSchema(
-            name=_get_bigquery_sanitized_field_name(key),
-            type=_get_bigquery_type_from_vcf_type(field.type),
-            mode=_TableFieldConstants.MODE_NULLABLE,
-            description=_get_bigquery_sanitized_field(field.desc)))
-
-  for annot_field in annotation_fields or []:
-    if not annot_field in header_fields.infos:
-      raise ValueError('Annotation field {} not found'.format(annot_field))
-    annotation_names = processed_variant.extract_annotation_names(
-        header_fields.infos[annot_field].desc)
-    annotation_record = bigquery.TableFieldSchema(
-        name=_get_bigquery_sanitized_field(annot_field),
-        type=_TableFieldConstants.TYPE_RECORD,
-        mode=_TableFieldConstants.MODE_REPEATED,
-        description='List of {} annotations for this alternate.'.format(
-            annot_field))
-    for annotation_name in annotation_names:
-      annotation_record.fields.append(bigquery.TableFieldSchema(
-          name=_get_bigquery_sanitized_field(annotation_name),
-          type=_TableFieldConstants.TYPE_STRING,
-          mode=_TableFieldConstants.MODE_NULLABLE,
-          # TODO(bashir2): Add descriptions of well known annotations, e.g.,
-          # from VEP.
-          description=''))
-    alternate_bases_record.fields.append(annotation_record)
-  schema.fields.append(alternate_bases_record)
+  proc_var_factory.add_alt_record_to_schema(schema)
```

Comment:
nit: instead of adding alt to a mutable schema, consider returning the alt schema and appending it to the schema object here (e.g. `schema.fields.append(proc_var_factory.get_alt_record_schema())`. This makes is more clear how the schema is being mutated in this function.

Reply:
Done.